connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/Plugins.java

```diff
@@ -234,6 +236,22 @@ public Converter newConverter(AbstractConfig config, String classPropertyName, C
         // Configure the Converter using only the old configuration mechanism ...
         String configPrefix = classPropertyName + ".";
         Map<String, Object> converterConfig = config.originalsWithPrefix(configPrefix);
+
+        // Have to override schemas.enable from true to false for internal JSON converters
+        // Don't have to warn the user about anything since all deprecation warnings take place in the
+        // WorkerConfig class
+        if (plugin instanceof JsonConverter) {
+            if (classPropertyName.equals(WorkerConfig.INTERNAL_KEY_CONVERTER_CLASS_CONFIG) ||
+                classPropertyName.equals(WorkerConfig.INTERNAL_VALUE_CONVERTER_CLASS_CONFIG)) {
+
+                // If they haven't explicitly specified values for internal.key.converter.schemas.enable
+                // or internal.value.converter.schemas.enable, we can safely default them to false
+                if (!converterConfig.containsKey(classPropertyName + "." + JsonConverterConfig.SCHEMAS_ENABLE_CONFIG)) {
+                    converterConfig.put(JsonConverterConfig.SCHEMAS_ENABLE_CONFIG, false);
```

Comment:
This setting doesn't seem to match the condition -- the condition checks for `internal.<x>.converter.schemas.enable` in `converterConfig`, but this is just setting `schemas.enable` in `converterConfig`. I think the condition check is the one that's wrong since `converterConfig` is generated via `originalsWithPrefix`